### PR TITLE
initial documentation code

### PR DESCRIPTION
**Pull request by Siddhant-Ray**
I am adding the inital ```docs``` folder to auto-build docs into the dev source tree.

Should not break anything as no source code changes within LMcache have been made.

### CODE REVIEW

**Comment by ApostaC** [comment]
Hey Siddhant, thanks for creating the PR! Will take a look soon

2 minor things:
- If you want to add reviewers, please add it in `reviewers` section instead of `assignees` section (see the screenshot below)
image
- Can you add "this PR resolves #<issue ID>" in the description if you think your PR addresses the issue? In this case, this PR will be linked to that issue and make it easier to track tasks. (Also, will be useful to take a look at this: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

**Comment by Siddhant-Ray** [comment]
Hey @ApostaC 

This initial PR doesn't completely close Issue #47 , as a few more iterations need to be made to have the documentation complete. So we shouldn't close the issue yet.

**Comment by ApostaC** [comment]
Got it!
So do you want to continue pushing to this PR, or merge it first and create different tickets for the remaining tasks?

**Comment by Siddhant-Ray** [comment]
Let's merge this one and I will create new, smaller tickets for specific parts of the documentation. It's too big to handle in one ticket I feel.

File: docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+build/
```
Comment (ApostaC) [inline docs/.gitignore]: Can you put it in the .gitignore in the root directory? Let's only have one gitignore

**Review by ApostaC — COMMENTED** [review]
Just got one more quick comment

File: docs/.gitignore (hunk 1 above)
Comment (Siddhant-Ray) [reply docs/.gitignore]: Resolved

**Review by ApostaC — APPROVED** [review]
Thanks, Siddhant. Looks good to me now. Let me merge it. 
Please create the tickets for the remaining documentation tasks

**Pull request merged**